source/reduce/merge_blocks_reduction_opportunity.cpp

```diff
@@ -34,9 +34,13 @@ MergeBlocksReductionOpportunity::MergeBlocksReductionOpportunity(
 }
 
 bool MergeBlocksReductionOpportunity::PreconditionHolds() {
-  // By construction, it is not possible for the merging of A->B to disable the
-  // merging of C->D, even when B and C are the same block.
-  return true;
```

Comment:
It would be good to have a comment explaining that there do exist scenarios where one merge can disable another merge.  Giving one example would help.

Reply:
Sounds good. Done.